src/classes/services/WeatherRestService.cls

```diff
@@ -5,43 +5,34 @@
 global with sharing class WeatherRestService {
 
     @HttpGet
-    global static Weather__c getById() {
-        RestRequest req = RestContext.request;
-        String weatherId = req.requestURI.substring(req.requestURI.lastIndexOf('/')+1);
-        Weather__c result = WeatherSelector.getById(weatherId);
-        return result;
-    }
-
-    @HttpGet
-    global static List<Weather__c> getByDateTime() {
-        RestRequest req = RestContext.request;
-        String weatherDateTime = req.requestURI.substring(req.requestURI.lastIndexOf('/')+1);
-        List<Weather__c> result = WeatherSelector.getByYear(weatherDateTime);
-        return result;
+    global static void getWeather() {
+        RestResponse res = RestContext.response;
```

Comment:
MAJOR
In case there is an error it would be great to store logs in Salesforce.

```
        Logger logger = new Logger();
        RestResponse res = RestContext.response;

        try {
            String body = WeatherService.handleRequest(RestContext.request);
            res.statusCode = 200;
            res.responseBody = Blob.valueOf(body);
        } catch (Exception e) {
            logger.createErrorLog(Summary.API, WeatherManager.class, e, res.statusCode);
        }
```

Logger class could have the method:

```
    public void createErrorLog(Summary summary, Type className, Exception e) {
        this.logs.add(
                new Log__c(
                        Name = className.getName(),
                        MethodName__c = e.getStackTraceString().substringBefore(':').substringAfterLast('.'),
                        LineNumber__c = e.getLineNumber(),
                        Severity__c = ApexPages.Severity.ERROR.name(),
                        Summary__c = String.valueOf(summary),
                        Message__c = e.getMessage(),
                        StackTrace__c = e.getStackTraceString()
                )
        );
    }
```

Create a new Log__c custom object to store logs as records, and take fields from the method above:
Name: auto-number
ClasName__c: Text 
MethodName__c = Text
LineNumber__c = Integer
Severity__c = Picklist (add values from ApexPages.Severity enum) 
Summary__c = Text
Message__c = Text Area Long
StackTrace__c = Text Area Long

Your Logger class can be simple with static method(s) to create Log__c records, or the same I posted above with private List<Log__c> that will be inserted in the end of consumer method exec:

```
        Logger logger = new Logger();
        ...

        try {
           ....
        } catch (Exception e) {
            logger.createErrorLog(Summary.API, WeatherManager.class, e, res.statusCode);
        } finally {
           logger.flush(); //inserts List<Log__c>
        }
```